floater/test/test_generators.py

```diff
@@ -53,6 +53,76 @@ def test_hexmesh(fs):
     # make sure things are "rectangular"
     assert np.allclose(x[0], x[1] + fs.dx/2)
 
+def test_oceancoords(fs):
+    """Verifies that the float grid excludes the masked regions properly."""
+
+    #excludes Anirban's example to speed up the testing time 
+    if fs.dx > 0.02:
+
+        #mask gird is half the spatial resolution of float set
+        mask_lon = np.linspace(fs.xlim[0], fs.xlim[1], int(fs.Nx/2))
+        mask_lat = np.linspace(fs.ylim[0], fs.ylim[1],int(fs.Ny/2))
+        mask = np.ones((np.shape(mask_lon)[0], np.shape(mask_lat)[0]))
+        
+
+        #fill top half of the domain with land 
+        coast_ind = int(len(mask_lat)/2) #index of coast 
+        coast_lat = mask_lat[coast_ind]
+        for j in range(np.shape(mask)[1]):
+            if j >= coast_ind:
+                mask[:,j] = 0
+        
+        #rect grid test 
+        float_x, float_y = np.transpose(fs.get_oceancoords(mesh='rect',mask=mask, mask_grid_x=mask_lon, mask_grid_y=mask_lat))
```

Comment:
Somehow it seems redundant to have to pass three separate masks to this function.